### ISSUE TITLE
Rejected file not available for reviewers

### ISSUE THREAD

**Issue by wagnerand** [mozilla/addons-server#8729]
https://addons.mozilla.org/firefox/downloads/file/997477/fuze_for_firefox-118.10615.12112.1-an+fx.xpi has been rejected but should still be available if you have reviewer permissions. Unfortunately, it shows 404 though.

**Comment by spoji** [comment]
Same thing is happening with 
https://addons.mozilla.org/firefox/downloads/file/1005816/robotext-2.7.6-an+fx.xpi?src=reviewer 

from review page https://reviewers.addons.mozilla.org/en-US/reviewers/review/robot-ext

**Comment by wagnerand** [comment]
> I strongly suspect that one is happening because the Addon is disabled by the developer (Invisible). We say "New version submissions for listing won't be accepted in this state." in devhub but the button is there so I guess we do.

Then that's a different issue. If the status was invisible before that version was uploaded, then the upload should have been prevented. If the status was set to invisible after the version was uploaded, it should the file should have been set to `STATUS_DISABLED`.

Also, for the record, the file viewer can access the file and display its contents, at least for this one.

**Comment by eviljeff** [comment]
>  If the status was invisible before that version was uploaded, then the upload should have been prevented.

it was set to invisible over a year ago https://addons.mozilla.org/en-GB/developers/feed/robot-ext

**Comment by eviljeff** [comment]
As for: 
> https://addons.mozilla.org/firefox/downloads/file/997477/fuze_for_firefox-118.10615.12112.1-an+fx.xpi 

kibana _seems_ to [indicate](https://amo-kibana.prod.mozaws.net/_plugin/kibana/#/doc/amo.*/amo.olympia_web.docker.addons-2018-07-05/amo.olympia_web.docker.addons?id=AWRqG02iC9_CWNk44Xb2&_g=()) that the `download_file` acl check is failing because `request.user.pk` is None "download file 997477: addon/file disabled and user None is not an owner or reviewer." 
https://github.com/mozilla/addons-server/blob/master/src/olympia/versions/views.py#L110:L122

This would happen if the user was anonymous which is weird.  **But** the number of logs in kibana don't match up to the number of times I've tried this, which leads me to believe the kibana log entries are just people trying the link from this github issue without being logged in.  So the 404 we're seeing is quite possibly not a django `raise http.404` but a genuine 404 because the file isn't there.

**Comment by eviljeff** [comment]
yeah it's certainly some intermittent weirdness because only that file is a 404 - the other similarly rejected files work fine.

**Comment by eviljeff** [comment]
Ops confirmed the file is in /addons, rather than /guarded_addons.  From looking in logs, the file was moved to /guarded-addons at 23:59:05
```
Timestamp:
    June 26th 2018, 23:59:05.000
Fields.msg:
    Moving disabled file: /mnt/efs/addons.mozilla.org/files/838624/fuze_for_firefox-118.10615.12112.1-an+fx.xpi => /mnt/efs/addons.mozilla.org/guarded-addons/838624/fuze_for_firefox-118.10615.12112.1-an+fx.xpi
```
but then moved back to /addons 23 seconds later:
```
Timestamp:
    June 26th 2018, 23:59:28.000
Fields.msg:
    Moving undisabled file: /mnt/efs/addons.mozilla.org/guarded-addons/838624/fuze_for_firefox-118.10615.12112.1-an+fx.xpi => /mnt/efs/addons.mozilla.org/files/838624/fuze_for_firefox-118.10615.12112.1-an+fx.xpi
```
What's not clear right now is why that happened, or why subsequent iterations of `hide_disabled_files` didn't move it back.
https://github.com/mozilla/addons-server/blob/master/src/olympia/addons/cron.py#L186:L221

**Comment by eviljeff** [comment]
https://github.com/mozilla/addons-server/blob/master/src/olympia/files/models.py#L320:L341 are the functions called to move the files around

**Comment by EnTeQuAk** [comment]
https://sentry.prod.mozaws.net/operations/olympia-prod/issues/3416954/

Is this error maybe related to the issue?

**Comment by eviljeff** [comment]
This morning, after @wagnerand had ops manually run `File.objects.get(id=997477).hide_disabled_file()` last night, I can download the file.

**Comment by wagnerand** [comment]
@eviljeff If there are no other ideas, I suggest that we add a log statement right before we call `hide_disabled_file` and another one in https://github.com/mozilla/addons-server/blob/21a1eddf3c738b77042a27cfba60d36945a70cf3/src/olympia/files/models.py#L323 or introduce an else block there to make sure `src` exists and is correct.

**Comment by wagnerand** [comment]
I am putting a few more examples in here for reference:

https://reviewers.addons.mozilla.org/en-US/reviewers/review/mixcloud-downloader v1.0.1: https://reviewers.addons.mozilla.org/en-US/firefox/files/browse/933519/

https://reviewers.addons.mozilla.org/en-US/reviewers/review/809812 v1.0.3.3: https://reviewers.addons.mozilla.org/en-US/firefox/files/browse/754254/

https://reviewers.addons.mozilla.org/en-US/reviewers/review/instant-idm-download-manager v1.0.1: https://reviewers.addons.mozilla.org/en-US/firefox/files/browse/925053/

https://reviewers.addons.mozilla.org/en-US/reviewers/review/open-via-idm-manager v1.0.1: https://reviewers.addons.mozilla.org/en-US/firefox/files/browse/867124/

**Comment by eviljeff** [comment]
> If there are no other ideas, I suggest that we add a log statement right before we call hide_disabled_file and another one in 'if storage.exists(src): ' or introduce an else block there to make sure src exists and is correct.

Logging would usually be the easiest method of debugging but there are 171k matches to that query ([redash](https://sql.telemetry.mozilla.org/queries/56935)) and the `if storage.exists()` is the way we test it's in the wrong place - i.e. we expect it to be false.  If we do something in the else like log again then that's another 171k log entries.  242,000 extra log entries extra 12 hours isn't desirable.

**Comment by wagnerand** [comment]
So, what is your suggestion?
